shared/src/business/useCases/docketEntry/fileCourtIssuedDocketEntryInteractor.js

```diff
@@ -31,128 +31,160 @@ exports.fileCourtIssuedDocketEntryInteractor = async (
     throw new Error('Docket entry has already been added to docket record');
   }
 
-  const user = await applicationContext
-    .getPersistenceGateway()
-    .getUserById({ applicationContext, userId: authorizedUser.userId });
-
   const numberOfPages = await applicationContext
     .getUseCaseHelpers()
     .countPagesInDocument({ applicationContext, docketEntryId });
 
-  const isUnservable = UNSERVABLE_EVENT_CODES.includes(documentMeta.eventCode);
+  const user = await applicationContext
+    .getPersistenceGateway()
+    .getUserById({ applicationContext, userId: authorizedUser.userId });
 
-  const docketEntryEntity = new DocketEntry(
-    {
-      ...omit(docketEntry, 'filedBy'),
-      attachments: documentMeta.attachments,
-      date: documentMeta.date,
-      documentTitle: documentMeta.generatedDocumentTitle,
-      documentType: documentMeta.documentType,
-      draftOrderState: null,
-      editState: JSON.stringify(documentMeta),
-      eventCode: documentMeta.eventCode,
-      filingDate: documentMeta.filingDate,
-      freeText: documentMeta.freeText,
-      isDraft: false,
-      isFileAttached: true,
-      isOnDocketRecord: true,
-      judge: documentMeta.judge,
-      numberOfPages,
-      scenario: documentMeta.scenario,
-      serviceStamp: documentMeta.serviceStamp,
-      trialLocation: documentMeta.trialLocation,
-      userId: user.userId,
-    },
-    { applicationContext },
-  );
-
-  const workItem = new WorkItem(
-    {
-      assigneeId: null,
-      assigneeName: null,
-      associatedJudge: caseToUpdate.associatedJudge,
-      caseStatus: caseToUpdate.status,
-      caseTitle: Case.getCaseTitle(caseEntity.caseCaption),
-      docketEntry: {
-        ...docketEntryEntity.toRawObject(),
-        createdAt: docketEntryEntity.createdAt,
-      },
-      docketNumber: caseToUpdate.docketNumber,
-      docketNumberWithSuffix: caseToUpdate.docketNumberWithSuffix,
-      hideFromPendingMessages: true,
-      inProgress: true,
-      section: DOCKET_SECTION,
-      sentBy: user.name,
-      sentByUserId: user.userId,
-    },
-    { applicationContext },
-  );
+  const isUnservable = UNSERVABLE_EVENT_CODES.includes(documentMeta.eventCode);
 
-  if (isUnservable) {
-    workItem.setAsCompleted({ message: 'completed', user });
-  }
+  for (let docketNumber of docketNumbers) {
+    const caseToUpdate = await applicationContext
+      .getPersistenceGateway()
+      .getCaseByDocketNumber({
+        applicationContext,
+        docketNumber,
+      });
 
-  docketEntryEntity.setWorkItem(workItem);
-  caseEntity.updateDocketEntry(docketEntryEntity);
+    let caseEntity = new Case(caseToUpdate, { applicationContext });
 
-  workItem.assignToUser({
-    assigneeId: user.userId,
-    assigneeName: user.name,
-    section: user.section,
-    sentBy: user.name,
-    sentBySection: user.section,
-    sentByUserId: user.userId,
-  });
+    const docketEntry = caseEntity.getDocketEntryById({
+      docketEntryId,
+    });
```

Comment:
Let's move this code lower and closer to where we use it in the `if(!docketEntry)` statement.